### DISCUSSION TITLE
How do I Pin a Maui App to a specific version?

### DISCUSSION THREAD

**Discussion by dmccolloughOneGas** [dotnet/maui#8886] [Ideas]
I created a new Maui App on my PC using VS2022 17.3.0 - Preview 3.0

When I open the solution and expand the dependicies I see that they are 6.0.408  
If I look at the maui releases page, I see that it corresponds to Maui SR 2 - 6.0.408

## Maui Releases

Maui GA - 6.0.312  
Maui SR1 - 6.0.400  
Maui SR2 - 6.0.408  
Maui SR3 - 6.0-.419

So if I wanted to rollback my new app to the GA version of Maui, I read that I should use  
a global.json file, so I tried adding one with the command
`dotnet new globaljson --sdk-version 6.0.312`

```
{
    "sdk": {
    "version": "6.0.312",
    "rollForward": "disable"
}
```

After reloading the solution, my dependencies are still at 6.0.408

I then tried running the command below based on a comment in this [url](https://github.com/dotnet/maui/issues/8066#issuecomment-1166325627).

`dotnet workload restore --from-rollback-file https://aka.ms/dotnet/maui/6.0.312.json --source https://aka.ms/dotnet6/nuget/index.json --source https://api.nuget.org/v3/index.json`

The output from the above command 'A compatible .NET SDK was not found.'

I read a comment from David Ortinau at this [link](https://github.com/dotnet/maui/discussions/6528#discussioncomment-2958998) that said Maui was tied to a .NET SDK version, if that's the case, **how are we supposed to know which .NET SDK version goes to which version of Maui?**

My intended outcome as I move forward with Maui in porting applications and creating new applications is that I know how to use a old version of Maui should a breaking change be introduced.

For example:  
      If my app is using Maui SR2 - 6.0.408 and I attempt to upgrade it to SR3 - 6.0.419  
      and for whatever reason SR3 horribly breaks my app and I need to revert to SR2.  
      **How do I go about doing this?**

I've spent day's trying to figure this out.  
So either I'm really stupid (I probably am) or this isn't as easy as I've seen alot of people making it out to be and I've seen alot of other people asking the same thing.

Perhaps maybe Gerald or James could make a nice video on how to accomplish this more maybe some straight forward documentation on how to make this work. A video, would be SUPER awesome!!!!!!!!

Thanks

**Comment by danch15** [comment]
https://github.com/dotnet/maui/discussions/8544
